packages/amplication-server/src/core/project/project.service.ts

```diff
@@ -114,13 +117,75 @@ export class ProjectService {
     return [...changedEntities, ...changedBlocks];
   }
 
+  async validateSubscriptionPlanLimitationsForProject(
+    projectId: string
+  ): Promise<void> {
+    const projects = await this.prisma.project.findMany({
+      where: {
+        id: projectId,
+      },
+      include: {
+        resources: {
+          include: {
+            entities: true,
+          },
+          where: {
+            resourceType: EnumResourceType.Service,
+            deletedAt: null,
+          },
+        },
+        workspace: {
+          include: {
+            subscriptions: true,
+            projects: {
+              include: {
+                resources: {
+                  where: {
+                    resourceType: EnumResourceType.Service,
+                    deletedAt: null,
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    });
+
+    const project = projects[0];
+    const workspace = project.workspace;
+    const subscriptions = workspace.subscriptions;
+    const workspaceServices = workspace.projects.flatMap(
+      (project) => project.resources
+    );
+    const projectServices = project.resources;
+
+    if (!subscriptions || subscriptions.length === 0) {
+      if (workspaceServices.length > LICENSE_SERVICES_PER_WORKSPACE_LIMIT) {
+        throw new Error(
```

Comment:
it is not an error only limitation of use. I would treat it as water flow response